src/ImageProcessorCore.Web/Middleware/ImageProcessorMiddleware.cs

```diff
@@ -0,0 +1,208 @@
+using ImageProcessorCore.Filters;
+using System;
+using System.Globalization;
+
+namespace ImageProcessorCore.Web.Middleware
+{
+    public class ImageProcessorMiddleware
+    {
+        private readonly IApplicationEnvironment _appEnvironment;
+
+        private readonly RequestDelegate _next;
+
+        public ImageProcessorMiddleware(RequestDelegate next, IApplicationEnvironment appEnvironment)
+        {
+            _next = next;
+            _appEnvironment = appEnvironment;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (!IsImageProcessorRequest(context))
+            {
+                // move to the next request here?
+                await _next.Invoke(context);
+                return;
+            }
+
+            int width = 0;
+            int.TryParse(context.Request.Query["width"], out width);
+            int height = 0;
+            int.TryParse(context.Request.Query["height"], out height);
+
+            var inputPath = _appEnvironment.ApplicationBasePath + "/wwwroot" + context.Request.Path;
+            using (var inputStream = File.OpenRead(inputPath))
+            {
+                var image = new Image(inputStream);
+
+                // write directly to the body output stream
+                using (var outputStream = new MemoryStream())
+                {
+                    //image.Resize(width, height)
+                    //    .Save(outputStream);
+
+                    var processors = GetImageProcessorFilters(context.Request.Query);
+                    if (width > 0 || height > 0)
+                    {
+                        image.Process(width, height, processors.ToArray())
+                            .Save(outputStream);
+                    }
+                    else
+                    {
+                        image.Process(processors.ToArray())
+                            .Save(outputStream);
+                    }
+
+                    var bytes = outputStream.ToArray();
+                    await context.Response.Body.WriteAsync(bytes, 0, bytes.Length);
+                }
+
+                //// the following approach will write to disk then serve the image
+                ////var inputPath = _appEnvironment.ApplicationBasePath + "/wwwroot" + context.Request.Path;
+                //var outputPath = _appEnvironment.ApplicationBasePath + "/Uploads/" + Path.GetFileName(context.Request.Path);
+                //using (var outputStream = File.OpenWrite(outputPath))
+                //{
+                //    image.Resize(width, height)
+                //        .Save(outputStream);
+                //}
+                //var bytes = File.ReadAllBytes(outputPath);
+                //await context.Response.Body.WriteAsync(bytes, 0, bytes.Length);
+            }
+
+            //await _next.Invoke(context);
+        }
+
+        private List<IImageProcessor> GetImageProcessorFilters(IReadableStringCollection queryString)
```

Comment:
While this works it's not extensible. In Framework a collection of `IWebGraphicsProcessor` are loaded via the `BuildManager` which allows for plugins. That behaviour should be copied and improved on.